x/incentive/keeper/rewards.go

```diff
@@ -0,0 +1,110 @@
+package keeper
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	cdptypes "github.com/kava-labs/kava/x/cdp/types"
+	"github.com/kava-labs/kava/x/incentive/types"
+)
+
+// HandleRewardPeriodExpiry deletes expired RewardPeriods from the store and creates a ClaimPeriod in the store for each expired RewardPeriod
+func (k Keeper) HandleRewardPeriodExpiry(ctx sdk.Context, rp types.RewardPeriod) {
+	k.CreateUniqueClaimPeriod(ctx, rp.Denom, rp.ClaimEnd, rp.ClaimTimeLock)
+	store := prefix.NewStore(ctx.KVStore(k.key), types.RewardPeriodKeyPrefix)
+	store.Delete(types.GetDenomBytes(rp.Denom))
+	return
+}
+
+// CreateNewRewardPeriod creates a new reward period from the input reward
+func (k Keeper) CreateNewRewardPeriod(ctx sdk.Context, reward types.Reward) {
+	// reward periods store the amount of rewards payed PER SECOND
+	rewardsPerSecond := sdk.NewDecFromInt(reward.AvailableRewards.Amount).Quo(sdk.NewDecFromInt(sdk.NewInt(int64(reward.Duration.Seconds())))).TruncateInt()
+	rewardCoinPerSecond := sdk.NewCoin(reward.AvailableRewards.Denom, rewardsPerSecond)
+	rp := types.RewardPeriod{
+		Denom:         reward.Denom,
+		Start:         ctx.BlockTime(),
+		End:           ctx.BlockTime().Add(reward.Duration),
+		Reward:        rewardCoinPerSecond,
+		ClaimEnd:      ctx.BlockTime().Add(reward.Duration).Add(reward.ClaimDuration),
+		ClaimTimeLock: reward.TimeLock,
+	}
+	k.SetRewardPeriod(ctx, rp)
+}
+
+// CreateAndDeleteRewardPeriods creates reward periods for active rewards that don't already have a reward period and deletes reward periods for inactive rewards that currently have a reward period
+func (k Keeper) CreateAndDeleteRewardPeriods(ctx sdk.Context) {
+	params := k.GetParams(ctx)
+
+	for _, r := range params.Rewards {
+		_, found := k.GetRewardPeriod(ctx, r.Denom)
+		// if governance has made a reward inactive, delete the current period
+		if !r.Active {
+			k.DeleteRewardPeriod(ctx, r.Denom)
+		}
+		// if a reward period for an active reward is not found, create one
+		if !found && r.Active {
+			k.CreateNewRewardPeriod(ctx, r)
+		}
```

Comment:
this can be replaced with a `switch` statement

Reply:
Bit confused here, is this what you are suggesting?
```
for _, r := range params.Rewards {
	switch _, found := k.GetRewardPeriod(ctx, r.Denom) {
	case found && !r.Active:
		k.DeleteRewardPeriod(ctx, r.Denom)
	case !found && r.Active:
		k.CreateNewRewardPeriod(ctx, r)
	}
}
```

Reply:
Couldn't figure out a way to elegantly use a switch statement here, lmk what you had in mind.